# women's criket 20-20 world cup 2023 match i am picking best 11 based on the performance of the world cup


it is end to end project 
1first i am extrac data from espn cricinfo website
2second store the that data to csv
3third cleaning the data
4fourth analysising the data
5fifth creating dashbord 

## first i am going to extrac data from espn website

''' These are important librarys to import for these project 
beautifulsoup4              used for web scraping
requests                    used for html file
csv                         used for csv converting
pandas                      used for dataframe work 

'''

In [1]:
import requests
from bs4 import BeautifulSoup
import csv
import pandas as pd

### extrac batting performance

In [2]:
#The first link all about most runs getter in the series 
#I used espn cricinfo website link

req=requests.get("https://stats.espncricinfo.com/ci/engine/records/batting/most_runs_career.html?id=15029;type=tournament")
#with the help of request i get html  data 
soup=BeautifulSoup(req.content,"html.parser")
#with the help of bs4 i parser the html file
table=soup.table # table is used here for store the table content
#print(table) if you to see that please print that
thead=table.find("thead").find_all('th') #using find function for find table head 
print(thead)



[<th class="left" nowrap="nowrap" title="player name">Player</th>, <th class="padAst" nowrap="nowrap" title="matches played">Mat</th>, <th nowrap="nowrap" title="innings batted">Inns</th>, <th nowrap="nowrap" title="not outs">NO</th>, <th nowrap="nowrap" title="runs scored"><b>Runs</b></th>, <th class="padAst" nowrap="nowrap" title="highest inns score">HS</th>, <th nowrap="nowrap" title="batting average">Ave</th>, <th nowrap="nowrap" title="balls faced">BF</th>, <th nowrap="nowrap" title="batting strike rate">SR</th>, <th nowrap="nowrap" title="hundreds scored">100</th>, <th nowrap="nowrap" title="fifties scored">50</th>, <th nowrap="nowrap" title="ducks scored">0</th>, <th nowrap="nowrap" title="boundary fours">4s</th>, <th nowrap="nowrap" title="boundary sixes">6s</th>]


In [3]:
table_head= [th.text for th in thead]     #using for loop to extrac the table head put that into list
print(table_head)

['Player', 'Mat', 'Inns', 'NO', 'Runs', 'HS', 'Ave', 'BF', 'SR', '100', '50', '0', '4s', '6s']


In [4]:
table_body = table.find("tbody").find_all("tr")    #here i am using find for find table body
print(table_body)

[<tr class="data1" data-days="738942">
<td class="left" nowrap="nowrap" title="record rank: 1"><a class="data-link" href="/ci/content/player/922481.html">L Wolvaardt</a> (SA-W)</td>
<td class="padAst" nowrap="nowrap">6</td>
<td nowrap="nowrap">6</td>
<td nowrap="nowrap">1</td>
<td nowrap="nowrap"><b>230</b></td>
<td nowrap="nowrap">66*</td>
<td nowrap="nowrap">46.00</td>
<td nowrap="nowrap">216</td>
<td nowrap="nowrap">106.48</td>
<td nowrap="nowrap">0</td>
<td nowrap="nowrap">3</td>
<td nowrap="nowrap">0</td>
<td nowrap="nowrap">22</td>
<td nowrap="nowrap">5</td>
</tr>, <tr class="data1" data-days="738940">
<td class="left" nowrap="nowrap" title="record rank: 2"><a class="data-link" href="/ci/content/player/515905.html">Nat Sciver-Brunt</a> (ENG-W)</td>
<td class="padAst" nowrap="nowrap">5</td>
<td nowrap="nowrap">5</td>
<td nowrap="nowrap">2</td>
<td nowrap="nowrap"><b>216</b></td>
<td nowrap="nowrap">81*</td>
<td nowrap="nowrap">72.00</td>
<td nowrap="nowrap">153</td>
<td nowrap="no

In [5]:
for tr in table_body:                         #i used for loop to extrac content rowise and put that in list
    table_data=[  td.text.strip() for td in tr.find_all("td")]  
    print(table_data)

['L Wolvaardt (SA-W)', '6', '6', '1', '230', '66*', '46.00', '216', '106.48', '0', '3', '0', '22', '5']
['Nat Sciver-Brunt (ENG-W)', '5', '5', '2', '216', '81*', '72.00', '153', '141.17', '0', '2', '0', '28', '2']
['BL Mooney (AUS-W)', '6', '6', '2', '206', '74*', '51.50', '175', '117.71', '0', '3', '1', '26', '2']
['AJ Healy (AUS-W)', '5', '5', '1', '189', '55', '47.25', '163', '115.95', '0', '2', '0', '25', '2']
['T Brits (SA-W)', '6', '6', '1', '186', '68', '37.20', '179', '103.91', '0', '2', '0', '19', '3']
['S Mandhana (IND-W)', '4', '4', '0', '151', '87', '37.75', '109', '138.53', '0', '2', '0', '18', '4']
['MM Lanning (AUS-W)', '6', '5', '2', '149', '49*', '49.66', '131', '113.74', '0', '0', '0', '16', '2']
['SW Bates (NZ-W)', '4', '4', '1', '137', '81*', '45.66', '113', '121.23', '0', '2', '2', '13', '1']
['RM Ghosh (IND-W)', '5', '5', '3', '136', '47*', '68.00', '104', '130.76', '0', '0', '1', '15', '2']
['HK Matthews (WI-W)', '4', '4', '1', '130', '66*', '43.33', '112', '116.

In [6]:
#here i am using csv to creat a report and store the table head and table body what i extra from website


with open("player_most_run_report(1).csv","w",newline="") as csv_file:
    csv_write=csv.writer(csv_file)
    csv_write.writerow(table_head)
    
    for tr in table_body:
        table_data=[  td.text.strip() for td in tr.find_all("td")]  
        csv_write.writerow(table_data)

In [8]:

tf=pd.read_csv("C:/Users/Kaleeshwaran D/Desktop/python learing/project/project criket analysis/player_most_run_report(1).csv")
tf.head()

Player  Mat  Inns  NO  Runs   HS    Ave   BF      SR  \
0        L Wolvaardt (SA-W)    6     6   1   230  66*  46.00  216  106.48   
1  Nat Sciver-Brunt (ENG-W)    5     5   2   216  81*  72.00  153  141.17   
2         BL Mooney (AUS-W)    6     6   2   206  74*  51.50  175  117.71   
3          AJ Healy (AUS-W)    5     5   1   189   55  47.25  163  115.95   
4            T Brits (SA-W)    6     6   1   186   68  37.20  179  103.91   

   100  50  0  4s  6s  
0    0   3  0  22   5  
1    0   2  0  28   2  
2    0   3  1  26   2  
3    0   2  0  25   2  
4    0   2  0  19   3

### extrac bowling performance

In [11]:
#table 2 most wciket
req=requests.get("https://stats.espncricinfo.com/ci/engine/records/bowling/most_wickets_career.html?id=15029;type=tournament")
#with the help of request i get html  data 
soup=BeautifulSoup(req.content,"html.parser")
#with the help of bs4 i parser the html file
table1=soup.table # table is used here for store the table content
#print(table) if you to see that please print that
thead1=table1.find("thead").find_all('th') #using find function for find table head 
print(thead1)

[<th class="left" nowrap="nowrap" title="player name">Player</th>, <th class="padAst" nowrap="nowrap" title="matches played">Mat</th>, <th nowrap="nowrap" title="innings bowled">Inns</th>, <th nowrap="nowrap" title="overs bowled">Overs</th>, <th nowrap="nowrap" title="maidens earned">Mdns</th>, <th nowrap="nowrap" title="runs conceded">Runs</th>, <th nowrap="nowrap" title="wickets taken"><b>Wkts</b></th>, <th nowrap="nowrap" title="best innings bowling">BBI</th>, <th nowrap="nowrap" title="bowling average">Ave</th>, <th nowrap="nowrap" title="economy rate">Econ</th>, <th nowrap="nowrap" title="bowling strike rate">SR</th>, <th nowrap="nowrap" title="four wkts in an inns">4</th>, <th nowrap="nowrap" title="five wkts in an inns">5</th>]


In [12]:
table_head1= [th.text for th in thead1] #using for loop to extrac the table head put that into list
print(table_head1)

['Player', 'Mat', 'Inns', 'Overs', 'Mdns', 'Runs', 'Wkts', 'BBI', 'Ave', 'Econ', 'SR', '4', '5']


In [13]:
table_body1 = table1.find("tbody").find_all("tr") #here i used find for find table body
print(table_body1)

[<tr class="data1" data-days="738940">
<td class="left" nowrap="nowrap" title="record rank: 1"><a class="data-link" href="/ci/content/player/878039.html">S Ecclestone</a> (ENG-W)</td>
<td class="padAst" nowrap="nowrap">5</td>
<td nowrap="nowrap">5</td>
<td nowrap="nowrap">20.0</td>
<td nowrap="nowrap">1</td>
<td nowrap="nowrap">83</td>
<td nowrap="nowrap"><b>11</b></td>
<td nowrap="nowrap">3/13</td>
<td nowrap="nowrap">7.54</td>
<td nowrap="nowrap">4.15</td>
<td nowrap="nowrap">10.9</td>
<td nowrap="nowrap">0</td>
<td nowrap="nowrap">0</td>
</tr>, <tr class="data1" data-days="738942">
<td class="left" nowrap="nowrap" title="record rank: 2"><a class="data-link" href="/ci/content/player/858809.html">A Gardner</a> (AUS-W)</td>
<td class="padAst" nowrap="nowrap">6</td>
<td nowrap="nowrap">6</td>
<td nowrap="nowrap">20.0</td>
<td nowrap="nowrap">0</td>
<td nowrap="nowrap">125</td>
<td nowrap="nowrap"><b>10</b></td>
<td nowrap="nowrap">5/12</td>
<td nowrap="nowrap">12.50</td>
<td nowrap="now

In [14]:
for tr in table_body1:     #i used for loop to extrac content rowise and put that in list
    table_data1=[  td.text.strip() for td in tr.find_all("td")]  
    print(table_data1)

['S Ecclestone (ENG-W)', '5', '5', '20.0', '1', '83', '11', '3/13', '7.54', '4.15', '10.9', '0', '0']
['A Gardner (AUS-W)', '6', '6', '20.0', '0', '125', '10', '5/12', '12.50', '6.25', '12.0', '0', '1']
['M Schutt (AUS-W)', '6', '6', '22.0', '0', '135', '10', '4/24', '13.50', '6.13', '13.2', '1', '0']
['M Kapp (SA-W)', '6', '6', '22.1', '0', '139', '9', '2/13', '15.44', '6.27', '14.7', '0', '0']
['LMM Tahuhu (NZ-W)', '4', '4', '15.0', '1', '95', '8', '3/37', '11.87', '6.33', '11.2', '0', '0']
['S Ismail (SA-W)', '6', '6', '22.0', '3', '129', '8', '3/27', '16.12', '5.86', '16.5', '0', '0']
['Nashra Sandhu (PAK-W)', '4', '4', '16.0', '0', '98', '7', '4/18', '14.00', '6.12', '13.7', '1', '0']
['D Brown (AUS-W)', '6', '6', '21.0', '1', '105', '7', '2/18', '15.00', '5.00', '18.0', '0', '0']
['A Khaka (SA-W)', '5', '5', '19.0', '0', '107', '7', '4/29', '15.28', '5.63', '16.2', '1', '0']
['S Glenn (ENG-W)', '5', '5', '18.0', '0', '109', '7', '3/19', '15.57', '6.05', '15.4', '0', '0']
['Renuka

In [15]:
#here i am using csv to creat a report and store the table head and table body what i extra from website


with open("player_best_bowling_report(2).csv","w",newline="") as csv_file:
    csv_write=csv.writer(csv_file)
    csv_write.writerow(table_head1)
    
    for tr in table_body1:
        table_data1=[  td.text.strip() for td in tr.find_all("td")]  
        csv_write.writerow(table_data1)

In [23]:

tf1=pd.read_csv("C:/Users/Kaleeshwaran D/Desktop/python learing/project/project criket analysis/player_best_bowling_report(2).csv")
tf1.head()

Player  Mat  Inns  Overs  Mdns  Runs  Wkts   BBI    Ave  \
0  S Ecclestone (ENG-W)    5     5   20.0     1    83    11  3/13   7.54   
1     A Gardner (AUS-W)    6     6   20.0     0   125    10  5/12  12.50   
2      M Schutt (AUS-W)    6     6   22.0     0   135    10  4/24  13.50   
3         M Kapp (SA-W)    6     6   22.1     0   139     9  2/13  15.44   
4     LMM Tahuhu (NZ-W)    4     4   15.0     1    95     8  3/37  11.87   

   Econ    SR  4  5  
0  4.15  10.9  0  0  
1  6.25  12.0  0  1  
2  6.13  13.2  1  0  
3  6.27  14.7  0  0  
4  6.33  11.2  0  0

In [17]:
#table 2 most wciket
req=requests.get("https://stats.espncricinfo.com/ci/engine/records/team/match_results.html?id=15029;type=tournament")
#with the help of request i get html  data 
soup=BeautifulSoup(req.content,"html.parser")
#with the help of bs4 i parser the html file
table2=soup.table # table is used here for store the table content
#print(table) if you to see that please print that
thead2=table2.find("thead").find_all('th') #using find function for find table head 
#print(thead1)
table_head2= [th.text for th in thead2] #using for loop to extrac the table head put that into list
#print(table_head2)
table_body2 = table2.find("tbody").find_all("tr") #here i used find for find table body
#print(table_body1)
for tr in table_body2:     #i used for loop to extrac content rowise and put that in list
    table_data2=[  td.text.strip() for td in tr.find_all("td")]  
    #print(table_data2)
#here i am using csv to creat a report and store the table head and table body what i extra from website


with open("match_summary_report(3).csv","w",newline="") as csv_file:
    csv_write=csv.writer(csv_file)
    csv_write.writerow(table_head2)
    
    for tr in table_body2:
        table_data2=[  td.text.strip() for td in tr.find_all("td")]  
        csv_write.writerow(table_data2)

In [22]:

tf2=pd.read_csv("C:/Users/Kaleeshwaran D/Desktop/python learing/project/project criket analysis/match_summary_report(3).csv")
tf.head()

Player  Mat  Inns  NO  Runs   HS    Ave   BF      SR  \
0        L Wolvaardt (SA-W)    6     6   1   230  66*  46.00  216  106.48   
1  Nat Sciver-Brunt (ENG-W)    5     5   2   216  81*  72.00  153  141.17   
2         BL Mooney (AUS-W)    6     6   2   206  74*  51.50  175  117.71   
3          AJ Healy (AUS-W)    5     5   1   189   55  47.25  163  115.95   
4            T Brits (SA-W)    6     6   1   186   68  37.20  179  103.91   

   100  50  0  4s  6s  
0    0   3  0  22   5  
1    0   2  0  28   2  
2    0   3  1  26   2  
3    0   2  0  25   2  
4    0   2  0  19   3

### based on batting and bowling performance i am creating allrounder table

In [ ]:
tf.set_index('Player', inplace = True) #set index as player

In [31]:
tf.head()   #this is batting performance 

Mat  Inns  NO  Runs   HS    Ave   BF      SR  100  \
Player                                                                        
L Wolvaardt (SA-W)          6     6   1   230  66*  46.00  216  106.48    0   
Nat Sciver-Brunt (ENG-W)    5     5   2   216  81*  72.00  153  141.17    0   
BL Mooney (AUS-W)           6     6   2   206  74*  51.50  175  117.71    0   
AJ Healy (AUS-W)            5     5   1   189   55  47.25  163  115.95    0   
T Brits (SA-W)              6     6   1   186   68  37.20  179  103.91    0   

                          50  0  4s  6s  
Player                                   
L Wolvaardt (SA-W)         3  0  22   5  
Nat Sciver-Brunt (ENG-W)   2  0  28   2  
BL Mooney (AUS-W)          3  1  26   2  
AJ Healy (AUS-W)           2  0  25   2  
T Brits (SA-W)             2  0  19   3

In [30]:
tf1.head()   #this is bowing performance 

Player  Mat  Inns  Overs  Mdns  Runs  Wkts   BBI    Ave  \
0  S Ecclestone (ENG-W)    5     5   20.0     1    83    11  3/13   7.54   
1     A Gardner (AUS-W)    6     6   20.0     0   125    10  5/12  12.50   
2      M Schutt (AUS-W)    6     6   22.0     0   135    10  4/24  13.50   
3         M Kapp (SA-W)    6     6   22.1     0   139     9  2/13  15.44   
4     LMM Tahuhu (NZ-W)    4     4   15.0     1    95     8  3/37  11.87   

   Econ    SR  4  5  
0  4.15  10.9  0  0  
1  6.25  12.0  0  1  
2  6.13  13.2  1  0  
3  6.27  14.7  0  0  
4  6.33  11.2  0  0

In [32]:
tf1.set_index('Player', inplace = True)  #set index as player
tf.head()

Mat  Inns  NO  Runs   HS    Ave   BF      SR  100  \
Player                                                                        
L Wolvaardt (SA-W)          6     6   1   230  66*  46.00  216  106.48    0   
Nat Sciver-Brunt (ENG-W)    5     5   2   216  81*  72.00  153  141.17    0   
BL Mooney (AUS-W)           6     6   2   206  74*  51.50  175  117.71    0   
AJ Healy (AUS-W)            5     5   1   189   55  47.25  163  115.95    0   
T Brits (SA-W)              6     6   1   186   68  37.20  179  103.91    0   

                          50  0  4s  6s  
Player                                   
L Wolvaardt (SA-W)         3  0  22   5  
Nat Sciver-Brunt (ENG-W)   2  0  28   2  
BL Mooney (AUS-W)          3  1  26   2  
AJ Healy (AUS-W)           2  0  25   2  
T Brits (SA-W)             2  0  19   3

### Cleaning the data and use inner merge to create allrounder performance table

In [38]:
allrounder = pd.merge(tf, tf1, left_index=True, right_index=True, how="inner") 

In [39]:
allrounder

Mat_x  Inns_x  NO  Runs_x   HS  Ave_x   BF    SR_x  \
Player                                                                      
HK Matthews (WI-W)         4       4   1     130  66*  43.33  112  116.07   
AC Kerr (NZ-W)             4       4   0     113   66  28.25  103  109.70   
A Gardner (AUS-W)          6       5   2     110   31  36.66   92  119.56   
O Prendergast (IRE-W)      4       4   0     109   61  27.25   87  125.28   
CL Tryon (SA-W)            6       5   0      79   40  15.80   73  108.21   
Nida Dar (PAK-W)           4       4   0      71   33  17.75   75   94.66   
CA Henry (WI-W)            4       4   0      61   34  15.25   55  110.90   
EA Perry (AUS-W)           6       4   1      60   40  20.00   37  162.16   
M Kapp (SA-W)              6       5   1      58  27*  14.50   48  120.83   
L Delany (IRE-W)           4       4   1      38  17*  12.66   52   73.07   

                       100  50  ...  Overs  Mdns  Runs_y  Wkts   BBI  Ave_y  \
Player                          ...                                           
HK Matthews (WI-W)       0   1  ...   14.0     1      89     4  2/14  22.25   
AC Kerr (NZ-W)           0   1  ...   14.5     0      75     6  3/23  12.50   
A Gardner (AUS-W)        0   0  ...   20.0     0     125    10  5/12  12.50   
O Prendergast (IRE-W)    0   1  ...   13.0     0      78     3  2/22  26.00   
CL Tryon (SA-W)          0   0  ...   10.0     0      71     3  2/12  23.66   
Nida Dar (PAK-W)         0   0  ...   15.0     1     101     5   2/5  20.20   
CA Henry (WI-W)          0   0  ...   13.3     1      99     3  2/30  33.00   
EA Perry (AUS-W)         0   0  ...    8.0     0      49     3   1/4  16.33   
M Kapp (SA-W)            0   0  ...   22.1     0     139     9  2/13  15.44   
L Delany (IRE-W)         0   0  ...   11.5     0      90     5  3/33  18.00   

                       Econ  SR_y  4  5  
Player                                   
HK Matthews (WI-W)     6.35  21.0  0  0  
AC Kerr (NZ-W)         5.05  14.8  0  0  
A Gardner (AUS-W)      6.25  12.0  0  1  
O Prendergast (IRE-W)  6.00  26.0  0  0  
CL Tryon (SA-W)        7.10  20.0  0  0  
Nida Dar (PAK-W)       6.73  18.0  0  0  
CA Henry (WI-W)        7.33  27.0  0  0  
EA Perry (AUS-W)       6.12  16.0  0  0  
M Kapp (SA-W)          6.27  14.7  0  0  
L Delany (IRE-W)       7.60  14.2  0  0  

[10 rows x 25 columns]

# here i am setting setting some parameter to segregate players based on preformance  

*first i segreagate batting performance 

*second i segreagate allrounder performance 

*third i segreagate bolwing performance


# setting player based on this parameters i am going to select top 5 best batting performance


1) Batting average  greater than 30  #average runs score in the innings

2) strick rate  greater than 110   #strick rate means run scored per ball 

3) innings batting  greater than 4 #number of match played in the series

4) boundary  percentage  #number of boundary scored by the player


In [ ]:
## setting player based on this parameters i am going to select top 3 allrounder performance

1) Batting average  greater than 25  #average runs score in the innings

2) strick rate  greater than 115   #strick rate means run scored per ball 

3) innings batting  greater and equla than 4 #number of match played in the series

4) boundary  percentage  #number of boundary scored by the player

5) bowling economics below 7  # run given by player as per over

6) bowling strike rate below 20  #taking wicket as per ball bolwed


In [ ]:
## setting player based on this parameters i am going to select top 3 bolwing  performance

1) innings bolwed  greater and equla than 4 #number of match played in the series

2) bowling economics below  than 6.5  # run given by player as per over

3) bowling strike rate below   than 16  #taking wicket as per ball bolwed

4) bowling averege below than 20 # a player's bowling average is the number of runs they have conceded per wicket taken

5) dot ball percentage    # number of dot balls bowled by the palyer
 